packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,582 @@ const double _kHandleSize = 22.0;
 const double _kToolbarScreenPadding = 8.0;
 const double _kToolbarHeight = 44.0;
 // Padding when positioning toolbar below selection.
-const double _kToolbarContentDistanceBelow = 16.0;
+const double _kToolbarContentDistanceBelow = _kHandleSize - 2.0;
 const double _kToolbarContentDistance = 8.0;
 
 /// Manages a copy/paste text selection toolbar.
-class _TextSelectionToolbar extends StatelessWidget {
+class _TextSelectionToolbar extends StatefulWidget {
   const _TextSelectionToolbar({
     Key key,
     this.handleCut,
     this.handleCopy,
     this.handlePaste,
     this.handleSelectAll,
+    this.isAbove,
   }) : super(key: key);
 
   final VoidCallback handleCut;
   final VoidCallback handleCopy;
   final VoidCallback handlePaste;
   final VoidCallback handleSelectAll;
 
+  // When true, the toolbar fits above its anchor and will be positioned there.
+  final bool isAbove;
+
+  @override
+  _TextSelectionToolbarState createState() => _TextSelectionToolbarState();
+}
+
+class _TextSelectionToolbarState extends State<_TextSelectionToolbar> with TickerProviderStateMixin {
+  // Whether or not the overflow menu is open. When it is closed, the menu
+  // items that don't overflow are shown. When it is open, only the overflowing
+  // menu items are shown.
+  bool _overflowOpen = false;
+
+  // True iff which menu items are shown has changed since the widget last
+  // updated.
+  bool _menuChanged = false;
```

Comment:
The `_menuChanged` business is strange because of edge-triggered vs. level-triggered. I wonder if we could do the following:
* remove this member.
* add a Key member variable, initialize it with a UniqueKey
* in didUpdateWidget, when we detect that the menu has changed, assign a new UniqueKey to this variable (and continue to close the menu).
* Use this key as a key for _TextSelectionToolbarContainer.

This way, when the menu changes substantially, we just build a new one from scratch.

Reply:
Done in https://github.com/flutter/flutter/pull/49391/commits/35774643eee558f108e619f5bcdf4304d65b494b, however, I still needed the `_menuChanged` variable within `_TextSelectionToolbarState` in order to include/exclude AnimatedSize.

Reply:
Interesting. I would have expected that due to the key change a new RenderAnimatedSize is instantiated and a new RenderAnimatedSize shouldn't be animating anything when it is first added. But that's not the case?